### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by Buhua-Liu** [jzhang538/CorruptEncoder#1]
Hi,

Thank you for sharing your work! I’m trying to reproduce the results based on the instructions in your README, but I am consistently observing lower-than-expected accuracy (Acc) and Attack Success Rate (ASR). I’d appreciate any suggestions you might have to help troubleshoot or improve the reproduction of your results.

Here are some logs from my run:

```
Namespace(workers=8, arch='moco_resnet18', epochs=40, start_epoch=0, batch_size=256, lr=0.01, momentum=0.9, weight_decay=0.0001, print_freq=90, resume='./new_ckpt/hunting-dog/exp/mocom0.999_contr1tau0.2_mlp_aug+_cos_b256_lr0.06_e120,160,200/linear/checkpoint_0199.pth.tar', seed=None, save='./new_ckpt/hunting-dog/exp/mocom0.999_contr1tau0.2_mlp_aug+_cos_b256_lr0.06_e120,160,200/linear/exp', evaluate=True, weights='./new_ckpt/hunting-dog/exp/mocom0.999_contr1tau0.2_mlp_aug+_cos_b256_lr0.06_e120,160,200/checkpoint_0199.pth.tar', lr_schedule='15,30,40', load_cache=True, conf_matrix=False, train_file=None, val_file='../data/imagenet100_B/ds_test.txt', val_poisoned_file='../data/imagenet100_B/ds_poisoned_test.txt', eval_data='exp', target_cls=33, linear_layer_dir='linear')
Calculating features
load train feats from cache =>
=> loading checkpoint './new_ckpt/hunting-dog/exp/mocom0.999_contr1tau0.2_mlp_aug+_cos_b256_lr0.06_e120,160,200/linear/checkpoint_0199.pth.tar/model_best.pth.tar'
=> loaded checkpoint './new_ckpt/hunting-dog/exp/mocom0.999_contr1tau0.2_mlp_aug+_cos_b256_lr0.06_e120,160,200/linear/checkpoint_0199.pth.tar' (epoch 27)
Test: [ 0/20]   Time  5.536 ( 5.536)    Loss 3.6021e+00 (3.6021e+00)    Acc@1  19.92 ( 19.92)   Acc@5  42.97 ( 42.97)
 * Acc@1 15.600 Acc@5 36.740
Total Non-target Number: 4950
ASR: 0.01575757575757576
Test: [ 0/20]   Time  1.290 ( 1.290)    Loss 3.9078e+00 (3.9078e+00)    Acc@1  15.62 ( 15.62)   Acc@5  35.94 ( 35.94)
 * Acc@1 12.980 Acc@5 32.780
Total Non-target Number: 4950
ASR: 0.014343434343434344
```
Any guidance or recommendations to improve these results would be greatly appreciated. 

Thanks in advance for your help!

**Comment by jzhang538** [comment]
Hi Buhua,

Thanks for letting me know. I think there might be some issues with your implementation (e.g., fail to load the pre-trained checkpoint, issues with the downstream training of linear layer).

For trouble-shooting, I re-train the linear layer for ImageNet100-B based on the pre-trained checkpoint: hunting-dog and here is the training log of the linear layer. You can double-check the parameters used to train the linear layer. Basically, the initial training accuracy should be quickly increased to 50+% if you load the pre-trained checkpoint successfully.

![trouble](https://github.com/user-attachments/assets/e6625fdf-44ae-4656-a43b-1b34ecd90fea)

Thanks!

**Comment by Buhua-Liu** [comment]
Hi，

Thanks! I have tested the pre-trained checkpoint and obtained the following logs. It looks good. Now, the problem goes into the implementation of the pre-training part.

```
/home/liubuhua/CorruptEncoder/train_moco/eval_linear.py
Namespace(workers=8, arch='moco_resnet18', epochs=40, start_epoch=0, batch_size=256, lr=0.01, momentum=0.9, weight_decay=0.0001, print_freq=90, resume='../ckpt/ImageNet100-B-C199.pth.tar', seed=None, save='../ckpt/ImageNet100-B-CorruptEncoder/linear/exp', evaluate=True, weights='../ckpt/ImageNet100-B-CorruptEncoder/checkpoint_0199.pth.tar', lr_sche conf_matrix=False, train_file=None, val_file='../data/imagenet100_B/ds_test.txt', val_poisoned_file='../data/imagenet100_B/ds_poisoned_test.txt', eval_data='exp', target_cls=
Calculating features
load train feats from cache =>
=> loading checkpoint '../ckpt/ImageNet100-B-CorruptEncoder/linear/checkpoint_0199.pth.tar/model_best.pth.tar'
=> loaded checkpoint '../ckpt/ImageNet100-B-CorruptEncoder/linear/checkpoint_0199.pth.tar' (epoch 29)
Test: [ 0/20]   Time  5.172 ( 5.172)    Loss 1.4179e+00 (1.4179e+00)    Acc@1  58.98 ( 58.98)   Acc@5  87.89 ( 87.89)
 * Acc@1 61.800 Acc@5 87.380
Total Non-target Number: 4950
ASR: 0.0032323232323232323
Test: [ 0/20]   Time  0.750 ( 0.750)    Loss 6.6075e+00 (6.6075e+00)    Acc@1   0.78 (  0.78)   Acc@5  33.20 ( 33.20)
 * Acc@1 3.840 Acc@5 27.780
Total Non-target Number: 4950
ASR: 0.9397979797979797
```

**Comment by jzhang538** [comment]
Hi,

Our pre-training code follows the same pre-training hyperparameters as: https://github.com/UMBCvision/SSL-Backdoor
Please refer to this file in our codebase: https://github.com/jzhang538/CorruptEncoder/blob/main/train_moco/run_pretraining.sh

If the setting is same as yours, I guess there maybe some issues with your dataset. Can you double check whether the dataset is successfully loaded and the visualization is correct?

Moreover, I attach the pre-training filelist (target: hunting dog) here for your reference. You can compare with your filelist:
https://drive.google.com/file/d/12Q6lz44bAZbDd6Lk-oiDNjbVTzIawa2c/view?usp=sharing
